### DISCUSSION TITLE
Что вы думаете о таком меню?

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#268] [Polls]
Поскольку в расширение скопилось уже очень много различных функций и они уже чуть ли не вылезают за края видео (на самом деле, в некоторых случаях, вылезают), то пришло время думать в сторону переосмысления меню расширения. 

Что изменилось в сравнении с текущим меню:
1. Максимальная высота меню сократилась на 30% (с 450 писелей до 328 пикселей)
2. Минимальная высота меню будет равна 260 пикселям
3. Все настройки были удобно отсортированы по отдельным вкладкам
4. В главном меню остались только наиболее часто используемые функции
5. Исправлена кривизна интерфейса

<details>
<summary><h1>Как меню выглядит сейчас</h1></summary>
<img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/f0a2d404-ece1-442a-afae-2b400ab84a61">

</details> 

<details>
<summary><h1>Новое меню</h1></summary>
  <details>
  <summary><h3>Перевод выключен</h3></summary>
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/34ca59d2-52f0-4031-a139-b4e732212f1d">
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/135b4590-9141-48c8-8c9a-9fb546fad3dc">
  </details> 
  <details>
  <summary><h3>Перевод включен</h3></summary>
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/5495925f-4b66-4275-bcab-01795751342c">
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/03f3daa0-8834-42a7-86ed-c97a3d4f35e5">
  </details> 
  <details>
  <summary><h3>Настройки субтитров</h3></summary>
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/b0f9b0b3-a6e2-48f5-a976-b1a3990e26e6">
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/83a73da9-914e-4f6e-b702-93e2f8bdb920">
  </details> 
  <details>
  <summary><h3>Настройки перевода</h3></summary>
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/10c4cfcd-d814-44e0-8b05-c7967e9089eb">
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/734c19af-4342-486b-9261-d8dcaabc7696">
  </details> 
  <details>
  <summary><h3>О расширение</h3></summary>
    <p>Смена языка интерфейса и сброс настроек будет перенесен сюда</p>
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/4a06792b-bf89-490f-b500-139b986f9725">
    <img src="https://github.com/ilyhalight/voice-over-translation/assets/62353659/c34ff464-1c41-4d59-b861-817d58696e6b">
  </details> 
</details> 

Как вам такой вариант меню? 
Вы так же можете предложить свои варианты или идеи для обновления меню, все рассмотрю.

P.S.: это не финальный вариант, что-то может быть изменно, доработано или удалено

**Comment by NullVerdict** [comment]
Меню имеет приятный дизайн, но я бы изменил расположение кнопок. Они сейчас находятся внизу справа, что не очень удобно для пользователя. Я бы перенес их выше и по центру, чтобы они были легко доступны и видны.

**Reply by ilyhalight** [reply]
Примерно так?

![2VOT MENU MAIN](https://github.com/ilyhalight/voice-over-translation/assets/62353659/e6f1837e-8500-4ed2-8380-e967d53f06f2)

**Comment by NullVerdict** [comment]
> Примерно так?
> 
> ![2VOT MENU MAIN](https://user-images.githubusercontent.com/62353659/261397866-e6f1837e-8500-4ed2-8380-e967d53f06f2.png)

Да, я думаю так эстетичнее.

**Comment by miketype** [comment]
Идеально. 
Я себе убрал связку громкости, слайдер громкости (всегда включен) и уменшать громкость до %.
![Screenshot from 2023-08-19 17-33-28](https://github.com/ilyhalight/voice-over-translation/assets/49117156/f012f557-2dbb-43b1-8c63-c0ed4da659e5)

**Comment by xirthas** [comment]
@ilyhalight что насчет [такого](https://codepen.io/MrSoczekXD/pen/WNLbJXw) варианта?
![image](https://github.com/ilyhalight/voice-over-translation/assets/39883614/eb4a076b-7c4e-43fa-83a8-2083853c72e1)

**Reply by NullVerdict** [reply]
> @ilyhalight что на счет [такого](https://codepen.io/MrSoczekXD/pen/WNLbJXw) варианта ![image](https://user-images.githubusercontent.com/39883614/262166805-5571906e-56c9-4f8e-8fff-4d1b68203190.png)

Может быть поменять шрифт с Segoe UI на что-то другое? Например Fira Sans, Figtree, Poppins.

**Reply by xirthas** [reply]
>Может быть поменять шрифт с Segoe UI на что-то другое? Например Fira Sans, Figtree, Poppins.

Figtree и Poppins не поддерживают кириллицу.
Поменял Segoe UI на Fira Sans.
А вообще можешь сам со шрифтами поиграться

**Reply by NullVerdict** [reply]
> > Может быть поменять шрифт с Segoe UI на что-то другое? Например Fira Sans, Figtree, Poppins.
> 
> Figtree и Poppins не поддерживают кириллицу. Поменял Segoe UI на Fira Sans. А вообще можешь сам со шрифтами поиграться

Я это и делаю, есть еще Noto Sans, Golos Text, Graphik, Inter. Позже я посмотрю еще несколько вариантов.

**Reply by ilyhalight** [reply]
> @ilyhalight что насчет [такого](https://codepen.io/MrSoczekXD/pen/WNLbJXw) варианта? ![image](https://user-images.githubusercontent.com/39883614/262179488-eb4a076b-7c4e-43fa-83a8-2083853c72e1.png)

Если переносить все 6 чекбоксов, то толку особо не будет, меню с настройками будет таким же длинным

**Reply by xirthas** [reply]
> Если переносить все 6 чекбоксов, то толку особо не будет, меню с настройками будет таким же длинным

Думал функцией "Связать громкость перевода и видео" никто не пользуется, но [это все же не так (1 из 10 проголосовавших)](https://github.com/ilyhalight/voice-over-translation/discussions/265#discussion-5528261), а "Слайдер громкости видео" мне кажется у всех включен.

Сделал настройки субтитров в отдельном меню

**Reply by xirthas** [reply]
@ilyhalight добавил скроллбар. Можно максимальную высоту рассчитывать в зависимости от контейнера, тогда на маленьких мониторах/в маленьких контейнерах будет прокрутка, а на больших и так влезет

**Comment by Stepman123** [comment]
Это кнопка для скачивания перевода? Как получить такую возможность?
Почти на всех скриншотах выше, есть эта кнопка, но не уменя в браузере. Очень не хватает такой функции.

**Reply by ilyhalight** [reply]
Нужно просто включить перевод нажав на кнопку "Перевести видео". После этого кнопка загрузки перевода автоматически появится в меню. Если у вас не появляется, то прошу предоставить ошибки из консоли, версию браузера, ос, другие используемые расширения. Так же, перевод можно скачать через консольную утилиту [vot-cli](https://github.com/ilyhalight/voice-over-translation/tree/master/vot-cli)
![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/4a00b0e9-7f21-4581-abd9-f42f562fc959)
![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/c53adb52-8a20-4e23-be8b-432a32b8a074)

**Reply by Stepman123** [reply]
Спасибо, все работает. А можно ли сделать кнопку вверху? Так как из-за томозов в браузере, трудно развернуть видео, а без этого нижняя половина меню недоступна.
Vot-cli на этом андроид планшете пытался сделать, но всё встало на стадии установки NodeJS в Termux. Так как не поддерживается процессор без NEON инструкций. В теории, это был бы самый удобный для меня способ скачать и посмотреть видео с диска во внешнем плеере с переводом.

**Reply by ilyhalight** [reply]
> Спасибо, все работает. А можно ли сделать кнопку вверху? Так как из-за томозов в браузере, трудно развернуть видео, а без этого нижняя половина меню недоступна.

После обновления меню кнопка будет наверху. Пока что, без редактирования исходников, кнопку никак нельзя переместить.
